src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -474,24 +475,27 @@ private File generateTargetFile(String originalFilename, String targetFolder, Bo
      * @param keepParentFolder    if true also creates the resources with the folder they are currently in (e.g. current/parent/* -> new/parent/*)
      * @throws IOException if the copying operation fails.
      */
-    public void copyResources(Resource[] resources, String prefix, String targetDirectoryPath, Boolean keepParentFolder) throws IOException {
+    public void copyResources(Resource[] resources, String prefix, Path targetDirectoryPath, boolean keepParentFolder) throws IOException {
         for (Resource resource : resources) {
             // Replace windows separator with "/"
-            String fileUrl = java.net.URLDecoder.decode(resource.getURL().toString(), StandardCharsets.UTF_8).replaceAll("\\\\", "/");
+            String fileUrl = java.net.URLDecoder.decode(resource.getURL().toString(), StandardCharsets.UTF_8).replace("\\\\", "/");
```

Comment:
Why don't we import the `URLDecoder`?
And why the change from replaceAll to replace?

Reply:
Using `replace` was suggested by @b-fein since we are not using a regex. Regarding the `URLDecoder` you're right one can simply import it and make the code less verbose